sonic_installer/main.py

```diff
@@ -208,6 +208,52 @@ def print_deprecation_warning(deprecated_cmd_or_subcmd, new_cmd_or_subcmd):
                 fg="red", err=True)
     click.secho("Please use '{}' instead".format(new_cmd_or_subcmd), fg="red", err=True)
 
+def update_sonic_environment(click, binary_image_version):
+    """Prepare sonic environment variable using incoming image template file. If incoming image template does not exist
+       use current image template file.
+    """
+    def mount_next_image_fs(squashfs_path, mount_point):
+        run_command("mkdir -p {0}".format(mount_point))
+        run_command("mount -t squashfs {0} {1}".format(squashfs_path, mount_point))
+
+    def umount_next_image_fs(mount_point):
+        run_command("umount -rf {0}".format(mount_point))
+        run_command("rm -rf {0}".format(mount_point))
+
+    SONIC_ENV_TEMPLATE_FILE = os.path.join("usr", "share", "sonic", "templates", "sonic-environment.j2")
+    SONIC_VERSION_YML_FILE = os.path.join("etc", "sonic", "sonic_version.yml")
+
+    sonic_version = re.sub("SONiC-OS-", '', binary_image_version)
+    new_image_dir = os.path.join('/', "host", "image-{0}".format(sonic_version))
+    new_image_squashfs_path = os.path.join(new_image_dir, "fs.squashfs")
+    new_image_mount = os.path.join('/', "tmp", "image-{0}-fs".format(sonic_version))
+
+    mount_next_image_fs(new_image_squashfs_path, new_image_mount)
+
+    next_sonic_env_template_file = os.path.join(new_image_mount, SONIC_ENV_TEMPLATE_FILE)
+    next_sonic_version_yml_file = os.path.join(new_image_mount, SONIC_VERSION_YML_FILE)
+
+    if not os.path.exists(next_sonic_env_template_file) or not os.path.exists(next_sonic_version_yml_file):
+        click.echo("SONiC environment file was not found on incoming image. Aborting...")
+        umount_next_image_fs(new_image_mount)
+        raise click.Abort()
+
+    try:
+        sonic_env = run_command_output(
```

Comment:
>run_command_output [](start = 20, length = 18)

Even this command failed (for example old image sonic-cfggen crash), it is safe to ignore and continue.